schema/3pse/file/applications/espresso/7.2/pw.x/system.json

```diff
@@ -325,27 +343,57 @@
         "Hubbard_alpha": {
             "type": "array",
             "items": {
-                "type": "number",
-                "default": 0
+                "type": "array",
+                "items": [
+                    {
+                        "type": "integer",
+                        "minimum": 1
+                    },
+                    {
+                        "type": "number",
+                        "default": 0
+                    }
+                ]
             }
         },
         "Hubbard_beta": {
             "type": "array",
             "items": {
-                "type": "number",
-                "default": 0
+                "type": "array",
+                "items": [
+                    {
+                        "type": "integer",
+                        "minimum": 1
+                    },
+                    {
+                        "type": "number",
+                        "default": 0
+                    }
+                ]
             }
         },
         "starting_ns_eigenvalue": {
             "type": "array",
             "items": {
                 "type": "array",
-                "items": {
-                    "type": "array",
-                    "items": {
+                "items": [
+                    {
+                        "type": "integer",
+                        "minimum": 1
+                    },
+                    {
+                        "type": "integer",
+                        "minimum": 1
+                    },
+                    {
+                        "type": "integer",
+                        "minimum": 1
+                    },
```

Comment:
We may encounter this type of value more often. Can we build reusable schemas, such as `tensor_index_3d`, `tensor_index_2d`, and `tensor_index_1d`? I'm just not entirely sure how we could extend the `"items"` array for different types...

Reply:
if we want to allow for some granular type safety here, then we cannot. While they will always be an array, the exact type limitations of each element vary for each parameter. Sometimes it will be an integer with minimum value of 1, but it is possible for it to be a float, or allow for negative numbers.